src/drivers/hda-dma.c

```diff
@@ -212,6 +212,14 @@ static int hda_dma_start(struct dma *dma, int channel)
 
 	/* full buffer is copied at startup */
 	p->chan[channel].desc_avail = p->chan[channel].desc_count;
+
+	/* for render let's wait for buffer full */
+	if (p->chan[channel].direction == DMA_DIR_HMEM_TO_LMEM) {
+		do {
+			idelay(PLATFORM_DEFAULT_DELAY);
+			dgcs = host_dma_reg_read(dma, channel, DGCS);
+		} while (!(dgcs & DGCS_BF));
```

Comment:
then how will you handle the pointers alignment with software buffer ones? I think you should do that in one same patch, otherwise, it will introduce noise issue again.

Reply:
What pointers alignment do you mean? BDL is much higher, than Host DMA buffer in FW. If there are some issues with Host DMA synchronization, you should consider using SPIB mode.